src/main/java/com/kenticocloud/delivery/DeliveryClient.java

```diff
@@ -350,6 +350,11 @@ protected HttpUriRequest buildGetRequest(String apiCall, List<NameValuePair> nam
     }
 
     protected RequestBuilder addHeaders(RequestBuilder requestBuilder) {
+        if (deliveryOptions.getProductionApiKey() != null) {
+            requestBuilder.setHeader(
+                    HttpHeaders.AUTHORIZATION, String.format("Bearer %s", deliveryOptions.getProductionApiKey())
+            );
+        }
         if (deliveryOptions.isUsePreviewApi()) {
             requestBuilder.setHeader(
                     HttpHeaders.AUTHORIZATION, String.format("Bearer %s", deliveryOptions.getPreviewApiKey())
```

Comment:
will this overwrite the authorization header if it was set already (above) or will it add it as a comma-separated value?

if B is true, i'd replace the if with else-if...

Reply:
I would have to test if this just replaces, or adds an additional header.  While implementing, I almost instinctively did if-else, but thought `If a consumer of the SDK explicitly said to set a production API key in the same client using the preview API, should we follow through with the request?`

This seems like incorrect usage of the SDK.  Perhaps we should generate an exception during construction or in the `DeliveryOptions`.  Do you have thoughts on this with knowing the internals of the API?

Reply:
@aweigold good point. yes, I think we should throw an exception if the developer intends to use both preview and secured delivery API. any other behavior would be non-transparent for the consumer developer.

@JanLenoch: could you please replicate this requirement into the other issues for other SDKs?